plugins/BEdita/Core/src/Model/Behavior/SearchableBehavior.php

```diff
@@ -29,84 +32,93 @@
 class SearchableBehavior extends Behavior
 {
     /**
-     * @inheritDoc
+     * {@inheritDoc}
+     *
+     * Deprecated configuration keys:
+     * - 'minLength' => 3,
+     * - 'maxWords' => 10,
+     * - 'columnTypes' => ['string', 'text'],
+     * - 'fields' => ['*' => 1]
+     *
+     * if present they are used in `SimpleAdapter` for backward compatibility.
      */
     protected $_defaultConfig = [
-        'minLength' => 3,
-        'maxWords' => 10,
-        'columnTypes' => [
-            'string',
-            'text',
-        ],
-        'fields' => [
-            '*' => 1,
-        ],
         'implementedFinders' => [
             'query' => 'findQuery',
         ],
     ];
 
     /**
-     * {@inheritDoc}
+     * The Search adapters registry instance.
+     *
+     * @var \BEdita\Core\Search\SearchRegistry
+     */
+    protected $searchRegistry = null;
+
+    /**
+     * Update search adapters index when a resource is saved.
      *
-     * If fields or column types are specified - do *not* merge them with existing config,
-     * overwrite the fields to search on.
+     * @param \Cake\Event\EventInterface $event The event
+     * @param \Cake\Datasource\EntityInterface $entity The resource entity
+     * @return void
      */
-    public function initialize(array $config): void
+    public function afterSave(EventInterface $event, EntityInterface $entity): void
     {
-        foreach (['columnTypes', 'fields'] as $key) {
-            if (isset($config[$key])) {
-                $this->setConfig($key, $config[$key], false);
-            }
+        foreach (array_keys((array)Configure::read('Search.adapters')) as $name) {
```

Comment:
Since this might become a common pattern should we extract a method `public function getAdapters(): string[]`?

Or perhaps a method to iterate through adapters altogether?

```php
/**
 * @return iterable<string, \BEdita\Core\Search\BaseAdapter>
 */
public function getAdapters(): iterable
{
    foreach (array_keys((array)Configure::read('Search.adapters')) as $name) {
        yield $name => $this->getAdapter($name);
    }
}
```